src/librustc_typeck/check/wfcheck.rs

```diff
@@ -411,6 +411,22 @@ fn check_type_defn<'tcx, F>(
                     ObligationCauseCode::MiscObligation,
                 )
             }
+
+            // Explicit `enum` discriminant values must const-evaluate successfully.
```

Comment:
Comment could use some elaboration regarding the "why" as well as a link towards https://github.com/rust-lang/rust/issues/70453.

Reply:
Just to make sure I'm following along, this concept of "must be evaluatable" is going to ultimately be true for *any* const expression, right? I think this is the basic WF criteria we are working towards?

Reply:
Yes, at the minimum any `ty::Const` used in the typesystem (array lengths and args to const params). While `enum` discrminants aren't that, they do require strict checks (no overlap), for which successful evaluatability is *a bare minimum*.